src/components/PostsList.tsx

```diff
@@ -2,53 +2,54 @@
 
 import classNames from 'classnames';
 import React from 'react';
-import { Post } from '../types/Post';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { actions as selectedPostActions } from '../features/selectedPost';
 
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
-
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+export const PostsList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { posts } = useAppSelector(state => state.author);
```

Comment:
Ensure that the `posts` array is always defined in the Redux state. If there's a possibility of it being undefined, consider adding a default value or a check before mapping over it.